kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyDeploymentDependentResource.java

```diff
@@ -69,26 +73,38 @@ protected Deployment desired(KafkaProxy primary,
                                  Context<KafkaProxy> context) {
         KafkaProxyContext kafkaProxyContext = KafkaProxyContext.proxyContext(context);
         var model = kafkaProxyContext.model();
+        String checksum = checksumFor(primary, context);
+
         // @formatter:off
         return new DeploymentBuilder()
                 .editOrNewMetadata()
                     .withName(deploymentName(primary))
                     .withNamespace(namespace(primary))
                     .addNewOwnerReferenceLike(ResourcesUtil.newOwnerReferenceTo(primary)).endOwnerReference()
                     .addToLabels(APP_KROXY)
-                    .addToLabels(standardLabels(primary))
-                .endMetadata()
-                .editOrNewSpec()
+                    .addToLabels(standardLabels(primary)).endMetadata().editOrNewSpec()
                     .withReplicas(1)
                     .editOrNewSelector()
                         .withMatchLabels(deploymentSelector(primary))
                     .endSelector()
-                    .withTemplate(podTemplate(primary, kafkaProxyContext, model.ingressModel(), model.clustersWithValidIngresses()))
+                    .withTemplate(podTemplate(primary, kafkaProxyContext, model.ingressModel(), model.clustersWithValidIngresses(), checksum))
                 .endSpec()
                 .build();
         // @formatter:on
     }
 
+    @VisibleForTesting
+    String checksumFor(KafkaProxy primary, Context<KafkaProxy> context) {
+        Optional<KafkaProxyIngress> kafkaProxyIngress = context.getSecondaryResource(KafkaProxyIngress.class);
+        MetadataChecksumGenerator checksumGenerator = context.managedWorkflowAndDependentResourceContext()
+                .get(MetadataChecksumGenerator.CHECKSUM_CONTEXT_KEY, MetadataChecksumGenerator.class)
+                .orElse(new Crc32ChecksumGenerator());
+
+        kafkaProxyIngress.ifPresent(checksumGenerator::appendMetadata);
```

Comment:
I'm confused by this.    I thought the deployment pod template would have the aggregrated checksum from the KafkaProxy.  I don't understand why it needs to consider a KafkaProxyIngress.

Reply:
Longer term yes, but this was a tactical choice to get something on the template earlier and be able to validate that the annotations were causing a rollout.

Reply:
Ok,  I see you did mention that on the PR.  Might be worth a comment in the code too.